index.js

```diff
@@ -20,6 +22,106 @@ app.get('/info', (req, res) => {
   res.send({ serverName: packageData.name, serverVersion: packageData.version });
 });
 
+// route for displaying all products
+app.get('/products/all', (req, res) => {
+
+  var jsonOutputObject = {
+    product_list_categorically : []
+  };
+
+  for(var i = 0; i < categoryData.categories.length; i++) {
+    
+    // JSON to be inserted in the main object
+    var jsonObjectToBeInserted = {
+      categoryId : categoryData.categories[i].id,
+      categoryName : categoryData.categories[i].categoryName,
+      products : []
+    }
+
+    for(var j = 0; j < productData.products.length; j++) {
+      if (productData.products[j].categoryId == categoryData.categories[i].id) {
+        jsonObjectToBeInserted.products.push(productData.products[j]);
+      }
+    }
+
+    jsonOutputObject.product_list_categorically.push(jsonObjectToBeInserted);
+  }
+
+  res.send(jsonOutputObject);
```

Comment:
The response of this would be categorized products according to your implementation

This API needs for getting all products as an Array including categoryName.

Reply:
Yeah, I was doubtful about that one, I thought combinig JSONs like that wouldn't make much sense so I tried to customize the implementation (I asked that on Telegram chat though), Anyway fixed it :)